src/main/java/bio/terra/service/iam/sam/SamRetry.java

```diff
@@ -33,40 +33,89 @@
         this.operationTimeout = now().plusSeconds(operationTimeoutSeconds);
     }
 
-    <T> T perform(SamFunction<T> function) throws InterruptedException {
+
+    @FunctionalInterface
+    public interface SamVoidFunction {
+        void apply() throws ApiException, InterruptedException;
+    }
+
+    @FunctionalInterface
+    public interface SamFunction<R> {
+        R apply() throws ApiException, InterruptedException;
+    }
+
+
+    static <T> T retry(ConfigurationService configService, SamFunction<T> function)
```

Comment:
I like this approach with the static!

Reply:
Thanks to @pshapiro4broad and @ddietterich for pointing me to it!